VoodooPS2Controller/VoodooPS2Controller.cpp

```diff
@@ -947,14 +957,7 @@ void ApplePS2Controller::setCommandByteGated(PS2Request* request)
 
 bool ApplePS2Controller::submitRequest(PS2Request * request)
 {
-  // Make sure that the request is in bounds
-  if (request->port >= kPS2MaxIdx)
-    return false;
-  
-  // Check that we only read from ports which are active in mux mode
-  if ((_mux_present && request->port == kPS2AuxIdx) ||
-      (!_mux_present && request->port > kPS2AuxIdx))
-    return false;
+  assert(request->port < kPS2MaxIdx);
```

Comment:
Simplification here. Requests for ports that don't devices that don't exist (but are under the max index) don't really cause issues. They'll just time out and are dumb to do. This really shouldn't happen anyways, so I don't think it's worth checking for. The port is filled in by the ApplePS2Mouse nubs, not the individual mouse driver.